lib/mercury/cps.rb

```diff
@@ -35,18 +35,18 @@ def run(*args, &k)
     #   Cps and returns a Cps
     def and_then(&pm)
       Cps.new do |*args, &k|
-        self.run(*args) do |*args2|
-          next_cps = pm.call(*args2)
-          next_cps.is_a?(Cps) or raise "'and_then' block did not return a Cps object. Did you want 'and_lift'? at #{pm.source_location}"
+        run(*args) do |*args2|
+          next_cps = yield(*args2)
+          next_cps.is_a?(Cps) || raise("'and_then' block did not return a Cps object. Did you want 'and_lift'? at #{pm.source_location}")
```

Comment:
`or` is more appropriate than `||` because I'm using it as a control construct (as opposed to a logical operator).